dask_expr/_collection.py

```diff
@@ -1242,9 +1243,17 @@ def set_index(
             if any([isinstance(c, FrameBase) for c in other]):
                 raise TypeError("List[FrameBase] not supported by set_index")
             elif not sorted:
-                raise NotImplementedError("Multi-column set_index requires sorted=True")
+                raise NotImplementedError(
+                    "Dask dataframe does not yet support multi-indexes.\n"
+                    f"You tried to index with this index: {other}\n"
+                    "Indexes must be single columns only."
+                )
```

Comment:
I'm a bit confused how this message relates to `sorted=False` and this new message. Previous message being "multi-column set_index requires stored=True" implies something different. The following error for `isinstance(other, DataFrame)` makes sense in that context.

Reply:
The new message is copied from dask/dask to stay consistent, the general idea is the following:

We support multi indexes only if we don't have to shuffle, e.g. if sorted=True, having a list of other would result in a MultiIndex